src/mixins/onThisPageRegistrator.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2021 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import { SectionKind } from 'docc-render/constants/PrimaryContentSection';
+import { anchorize } from 'docc-render/utils/strings';
+import {
+  MainContentSectionAnchors,
+  PrimaryContentSectionAnchors,
+} from 'docc-render/constants/ContentSectionAnchors';
+
+/**
+ * Crawls the `topicData` of a page, and extracts onThisPage sections.
+ */
+export default {
+  watch: {
+    topicData: {
+      immediate: true,
+      handler: 'extractOnThisPageSections',
+    },
+  },
+  methods: {
+    shouldRegisterContentSection(subSection) {
+      return subSection.type === 'heading' && subSection.level < 4;
+    },
+    extractOnThisPageSections(topicData) {
+      if (!topicData) return;
+      // unregister old items
+      this.store.resetPageSections();
+      // register new ones
+      const {
+        primaryContentSections,
+        topicSections,
+        defaultImplementationsSections,
+        relationshipsSections,
+        seeAlsoSections,
+      } = topicData;
+      if (primaryContentSections) {
+        primaryContentSections.forEach((section) => {
+          switch (section.kind) {
+          case SectionKind.content:
+            section.content.forEach((subSection) => {
```

Comment:
Maybe it's not _strictly_ necessary since headings will generally tend to be at the top-most level of the "content tree" 95% of the time, but this could potentially miss items since it only loops over the top level array of nodes and doesn't do a recursive loop over the full tree.

If you want, you could potentially just import the `ContentNode.methods.forEach` function which will do a recursive loop through the tree nodes (just might to use `bind` to change the context to work with this content array—there may be an example of where this was done in the past).